clusterous/cluster.py

```diff
@@ -780,49 +992,52 @@ def init_cluster(self, cluster_name, cluster_spec, nodes_info=[], logging_level=
                 self._logger.debug('Shared volume {0} created'.format(shared_vol.id))
                 conn.create_tags([shared_vol.id], {'Name': defaults.controller_name_format.format(cluster_name),
                                                    defaults.instance_tag_key: cluster_name})
-
+    
                 attach = shared_vol.attach(controller_res.instances[0].id, '/dev/sdf')
                 while shared_vol.attachment_state() != 'attached':
                     time.sleep(2)
                     shared_vol.update()
-
+    
             # Tag central logging
             central_logging = {}
             if logging_tags_and_res:
                 self._logger.debug('Tagging central logging...')
                 central_logging = self._wait_and_tag_instance_reservations(logging_tags_and_res)
-
+    
             # Tag all nodes
             nodes = {}
             if node_tags_and_res:
                 nodes = self._wait_and_tag_instance_reservations(node_tags_and_res)
-
-            # Any errors that occur up until this point cannot be recovered from by destroy
+    
+             # Any errors that occur up until this point cannot be recovered from by destroy
         except (Exception, KeyboardInterrupt) as e:
-            raise ClusterException('An error occured during cluster creation. Any created AWS EC2 instances will have to be terminated manually')
-
+             raise ClusterException('An error occured during cluster creation. Any created AWS EC2 instances will have to be terminated manually')
+     
         try:
             # Extra variables used by ansible scripts
             extra_vars = {'central_logging_level': logging_level,
                           'central_logging_ip': '',
                           'byo_volume': 1 if shared_volume_id else 0
                           }
-
+    
             # Configure controller
             controller_vars_dict = self._controller_vars_dict()
             controller_vars_dict.update(extra_vars)
             controller_vars_file = self._make_vars_file(controller_vars_dict)
-
+    
+    
+            time.sleep(30)
```

Comment:
Again, why is there a sleep here? If we need to wait for something, we should ideally poll until we're ready to proceed. Putting a sleep should be the last resort, only used if other options do not work properly.